torch/quantization/observer.py

```diff
@@ -335,18 +335,20 @@ def extra_repr(self):
 
     def _save_to_state_dict(self, destination, prefix, keep_vars):
         super(MinMaxObserver, self)._save_to_state_dict(destination, prefix, keep_vars)
-        destination[prefix + 'min_val'] = self.min_val
-        destination[prefix + 'max_val'] = self.max_val
+        if self.min_val is not None and self.max_val is not None:
+            destination[prefix + 'min_val'] = self.min_val
+            destination[prefix + 'max_val'] = self.max_val
 
     def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                               missing_keys, unexpected_keys, error_msgs):
 
         local_state = ['min_val', 'max_val']
+        optional_keys = ['min_val', 'max_val']
         for name in local_state:
             key = prefix + name
             if key in state_dict:
                 setattr(self, name, state_dict.pop(key))
-            elif strict:
+            elif strict and not key in optional_keys:
                 missing_keys.append(key)
         super(MinMaxObserver, self)._load_from_state_dict(state_dict, prefix, local_metadata, strict,
                                                           missing_keys, unexpected_keys, error_msgs)
```

Comment:
@raghuramank100 these seem to be repeated for all observers can we move them to `_ObserverBase`?